### DISCUSSION TITLE
How to center text within a sprite

### DISCUSSION THREAD

**Discussion by filipe-costa** [pixijs/pixijs#7932] [Q&A]
Hello all,

I understand that this might be a simple question and probably this was already answered in html5gamedevs.com forum, which is unfortunately down due to security issues.

I have created a [minimum reproducible demo](https://github.com/filipe-costa/pixijs-demo-text) regarding this matter.

For context, I am trying to use a sprite as a background and I would like to have some text that would be in the center of the sprite. I have tried to set the anchor of both container, sprite and text but to no avail.

On the following screenshot, I have set the anchor of the right sprite text to 0.5 (text.anchor.set(0.5)), which I think it is or was supposed to center the text.

![Screenshot 2021-10-30 at 23 45 45](https://user-images.githubusercontent.com/25135035/139560526-7e85ef36-4c29-44be-959c-ca7ffaf74442.png)

What are the possible workarounds or best practices related to this?

**Comment by kohillyang** [comment]
Just call addChild since sprite is a Container. Like the following codes:
```Javascript
sprite.addChild(text)
text.x = .5 * sprite.width
text.y = .5 * sprite.height
```
